src/callbacks/callbacks.jl

```diff
@@ -434,6 +434,64 @@ function save_restart_func(integrator, output_dir)
     return nothing
 end
 
+Base.@kwdef mutable struct WallTimeEstimate
+    n_calls::Int = 0
+    n_next::Int = 1
+    t_last::Float64 = -1
+    ∑Δt_wall::Float64 = 0
+end
+import Dates
+function print_walltime_estimate(integrator)
+    (; walltime_estimate, dt, t_end) = integrator.p
+    wte = walltime_estimate
+    # 1st call initializes `t_last` (during integrator init)
+    # 2nd call computes `Δt` for the first time (but also includes compilation), so we skip
+    cond = wte.n_calls > 1
+    if cond
+        # We need to account for skipping cost of `Δt` when `n_calls == 1`:
+        factor = wte.n_calls == 2 ? 2 : 1
+        Δt = factor * (time() - wte.t_last)
+    else
+        wte.n_calls == 1 && @info "Progress: Completed first step"
+        Δt = Float64(0)
+        wte.n_next = wte.n_calls + 1
+    end
+    wte.∑Δt_wall += Δt
+    wte.t_last = time()
+
+    if wte.n_calls == wte.n_next && cond
+        t = integrator.t
+        n_steps_total = ceil(Int, t_end / dt)
+        n_steps = ceil(Int, t / dt)
+        t_wall_ave_per_step = wte.∑Δt_wall / n_steps
+        twps = t_wall_ave_per_step
+        _twps = time_and_units(t_wall_ave_per_step)
+        percent_complete = round(t / t_end * 100; digits = 1)
+        n_steps_remaining = n_steps_total - n_steps
+        erwt = twps * n_steps_remaining
+        _erwt = time_and_units(erwt)
+        etwt = time_and_units(twps * n_steps_total)
+        spent = time_and_units(wte.∑Δt_wall)
+        t_sim = time_and_units(Float64(t))
+        sypd = round(
+            simulated_years_per_day(
+                EfficiencyStats((zero(t), t), wte.∑Δt_wall),
+            );
+            digits = 3,
+        )
+        estimated_finish_date =
+            Dates.now() + compound_period(erwt, Dates.Second)
+        @info "Progress" simulation_time = t_sim n_steps_completed = n_steps wall_time_per_step =
+            _twps wall_time_total = etwt wall_time_remaining = erwt wall_time_spent =
+            spent percent_complete = "$percent_complete%" sypd = sypd date_now =
+            Dates.now() estimated_finish_date = estimated_finish_date
+        wte.n_next *= 2 # doubling factor (to reduce log noise)
```

Comment:
I think we should have a cap (say, every 10 %), so that we still get periodic logs after 50 % of the simulation.

Reply:
Yeah, I guess that's not a bad idea. It should only add another, at most, 10 more printed logs.